MADRaS/utils/gym_torcs.py

```diff
@@ -195,15 +198,32 @@ def reset(self, client, relaunch=False):
         self.initial_reset = False
         return self.get_obs(), client
 
-    def end(self):
-        os.system('pkill torcs')
+    def end(self,client):
+        command = 'kill {}'.format(client.serverPID)
+        
+        os.system(command)
 
     def get_obs(self):
         return self.observation
 
-    def reset_torcs(self):
-        print("relaunch torcs")
-        os.system('pkill torcs')
+    def reset_torcs(self,client):
+        print("relaunch torcs in gym_torcs on port{}".format(client.port))
+        
+        command = 'kill {}'.format(client.serverPID)
+        os.system(command)
+        
+        command = None
+        rank = MPI.COMM_WORLD.Get_rank()        
+        if rank == 0 and self.visualise:
```

Comment:
As discussed offline, instead of rank == 0 , introduce a new variable, which could be set by the user.